src/hdmf/build/objectmapper.py

```diff
@@ -1254,8 +1254,10 @@ def construct(self, **kwargs):
 
     def __new_container__(self, cls, container_source, parent, object_id, **kwargs):
         """A wrapper function for ensuring a container gets everything set appropriately"""
-        obj = cls.__new__(cls, container_source=container_source, parent=parent, object_id=object_id)
+        obj = cls.__new__(cls, container_source=container_source, parent=parent, object_id=object_id,
+                          in_construct_mode=True)
```

Comment:
```suggestion
        obj = cls.__new__(cls, container_source=container_source, parent=parent, object_id=object_id,
                          in_construct_mode=True)
        # obj has been created and is in construction mode, indicating that the object is being constructed by
        # the automatic build process during read, rather than by the user
```